### DISCUSSION TITLE
Merged twins in titanium

### DISCUSSION THREAD

**Discussion by PengWei97** [mtex-toolbox/mtex#1513] [Ask Anything] [answered]
Dear MTEX experts,

Recently I want to merge tensile twins {11-22} in pure titanium, but I don't know how to set `orientation.map`. 

Specifically, I have merged some of the twins according to the tutorial - [Merging Grains](https://mtex-toolbox.github.io/GrainMerge.html), but there are still some twins that have not been merged successfully. As shown in the red ellipse in the figure below,
![image](https://user-images.githubusercontent.com/56183775/182066042-b9d9ab23-9701-4e8e-a987-0ddffbc6f984.png)

Also, I know that the red ellipse box selects the Euler angle of the twin, which is Euler = (71.3°, 81.6°, 213.5°). But I don't know how to get the four parameters (u1, v1, u2, v2) required by `orientation.map` through `round2Miller`, which is not mentioned in #975.

Finally, here is the code I ran and the ebsd raw data,

```matlab
clear all
close all
% 
my_linewidth = 0.5;
% crystal symmetry
CS = {... 
  'notIndexed',...
  crystalSymmetry('6/mmm', [3 3 4.7], 'X||a*', 'Y||b', 'Z||c*', 'mineral', 'Ti-Hex', 'color', [0.53 0.81 0.98])};

% plotting convention
setMTEXpref('xAxisDirection','north');
setMTEXpref('zAxisDirection','outOfPlane');

%% Specify File Names

% path to files
pname = 'E:\Matlab\temp';

% which files to be imported
fname = [pname '\twin.ctf'];

%% Import the Data

% create an EBSD variable containing the data
ebsd = EBSD.load(fname,CS,'interface','ctf',...
  'convertEuler2SpatialReferenceFrame');
plotx2east

close all
my_linewidth = 1.0;
figure(1)
[grains,ebsd.grainId,ebsd.mis2mean] = calcGrains(ebsd,'threshold',1.0*degree);
ebsd(grains(grains.grainSize<20)) = [];
notIndexed = grains('notIndexed');

% the "not indexed grains" we want to remove
toRemove = notIndexed(notIndexed.grainSize ./ notIndexed.boundarySize<2.0);

% and perform grain reconstruction with the reduces EBSD data set
[grains,ebsd.grainId,ebsd.mis2mean] = calcGrains(ebsd,'threshold',1.0*degree);
% [grains,ebsd.grainId,ebsd.mis2mean] = calcGrains(ebsd);
grains = smooth(grains,50);


% define twinning misorientation
CS = grains.CS;
twinning = orientation.map(Miller(0,1,-1,-2,CS),Miller(0,-1,1,-2,CS),...
  Miller(2,-1,-1,0,CS),Miller(2,-1,-1,0,CS));
  
% extract all Ti Magnesium grain boundaries
gB = grains.boundary('Ti-Hex','Ti-Hex'); 
% plot(ebsd,ebsd.orientations,'micronbar','on','coordinates','on');
% hold on
plot(grains.boundary);

% and check which of them are twinning boundaries with threshold 5 degree
isTwinning = angle(gB.misorientation,twinning) < 10*degree;
twinBoundary = gB(isTwinning)
[mergedGrains,parentId] = merge(grains,twinBoundary);

plot(mergedGrains.boundary, 'lineWidth',my_linewidth,'micronbar','off','coordinates','off');
% plot(mergedGrains)
```
[twin.zip](https://github.com/mtex-toolbox/mtex/files/9230702/twin.zip)

Any suggestions or recommendations to fix the problem would be greatly appreciated.

Thank you
Wei

**Comment by kilir** [comment, marked as answer]
Hi you can define your twinning misorientation by any means e.g. `orientation.byEuler` `orientation.byAxisAngle`, etc... no need to do it by ´orientation.map´. The only requirement is, that you need to know it. 

Some people approach the lack of a previously known twinning misorientation by searching for those occupying local maxima in an MDF and fitting using `round2Miller`. However, if you know it, it's not needed at all.

Hope this helps.

Cheers,
Rüdiger

**Reply by PengWei97** [reply]
Do I have to replace `twinning = orientation.map(Miller(0,1,-1,-2,CS),Miller(0,-1,1,-2,CS),Miller(2,-1,-1,0,CS),Miller(2,-1,-1,0,CS));` with `twinning = orientation.byEuler(71.3*degree,81.6*degree,213.5 *degree, CS)`?

**Reply by PengWei97** [reply]
Thank you so much. This issue has been properly resolved based on your suggestion. 

However, the problem I have now is that I don't know how to get the new ebsd based on the existing `mergedGrains`, and a new discussion (#1657) has been created, looking forward to your further suggestions.

peng
